### PR TITLE
improve validations for credit and debit models

### PR DESCRIPTION
**Pull request by bryanchriswhite**

### CODE REVIEW

**Comment by phutchins** [comment]
Looks good. Merging.

**Review by phutchins — APPROVED** [review]
Looks good.

File: test/debit.unit.js [hunk 1]
```diff
@@ -97,8 +97,9 @@ describe('Storage/models/Debit', function() {
         amount: null
       });
 
-      newDebit.save(function(err, debit) {
-        expect(debit.amount).to.equal(0);
+      newDebit.save(function(err) {
+        expect(err).to.be.instanceOf(Error);
```
Comment (braydonf) [inline test/debit.unit.js:101]: nit: the error messages here are not clear, using this instead gives better messages
```
if (err) {
  return done(err)
}
```
There are other places like this that exist already, but FYI.

**Comment by braydonf** [comment]
It would be useful to have test cases for this branch, but otherwise looks fine.

File: test/debit.unit.js (hunk 1 above)
Comment (bryanchriswhite) [reply test/debit.unit.js:101]: @braydonf when you say "...gives better error messages" do you mean in the test output in the event of failure?

Comment (braydonf) [reply test/debit.unit.js:101]: Wait.... nevermind. This is checking for an error, not the opposite.

**Comment by bryanchriswhite** [comment]
@braydonf do you mean adding test cases to exercise the getters and setters?

**Comment by braydonf** [comment]
Yeah, how is the case triggered?

I'm assuming it is because: 
```
> Math.round(undefined * 10000)
NaN
```

**Comment by bryanchriswhite** [comment]
Yes but also null

**Comment by braydonf** [comment]
On another note, using `Number.isInteger()` or `Number.isSafeInteger()` may be useful to validate values. I'm not sure what default validation is done with `Number` type in mongoose.

**Comment by bryanchriswhite** [comment]
@braydonf cool - i'll keep that in mind. For now tho this resolved the issue i had while testing in staging and we want to get this out asap. Any objections to merging?

**Comment by phutchins** [comment]
Looks like the discussion is down to semantics so I'm going to go ahead and merge this as to not hold up the assembly line... We can discuss any cleanup or consistency separately.

**Pull request merged**